### PR TITLE
Implement unit tests for AnomalyDetectorsIndex class

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR adds unit tests for `AnomalyDetectorsIndex` class.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    public void testCreateStateIndexAndAliasIfNecessary_WriteAliasAlreadyExistsAndPointsAtSubsequentStateIndex() {
+        testCreateStateIndexAndAliasIfNecessary_WriteAliasAlreadyExists(".ml-state-000007");
+    }
+
+    public void testCreateStateIndexAndAliasIfNecessary_WriteAliasAlreadyExistsAndPointsAtDummyIndex() {
+        testCreateStateIndexAndAliasIfNecessary_WriteAliasAlreadyExists("dummy-index");
+    }
+
+    private void testCreateStateIndexAndAliasIfNecessary_WriteAliasDoesNotExist(List<String> existingIndexNames,
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java]: nit: its confusing to prefix with `test` as this is usually the entry point (if public) for a unit test. I tend towards naming inner methods like this `assert...`

**Review by hendrikmuhs — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    private ArgumentCaptor<CreateIndexRequest> createRequestCaptor;
+    private ArgumentCaptor<IndicesAliasesRequest> aliasesRequestCaptor;
+
+    @Before
+    public void setUpMocks() {
+        threadPool = mock(ThreadPool.class);
+        when(threadPool.getThreadContext()).thenReturn(new ThreadContext(Settings.EMPTY));
+
+        indicesAdminClient = mock(IndicesAdminClient.class);
+        when(indicesAdminClient.prepareCreate(".ml-state"))
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java]: Done.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndexTests.java]: Done.

**Pull request merged**